src/connection.c

```diff
@@ -389,7 +389,11 @@ int connGetSocketError(connection *conn) {
 }
 
 int connPeerToString(connection *conn, char *ip, size_t ip_len, int *port) {
-    return anetFdToString(conn ? conn->fd : -1, ip, ip_len, port, FD_TO_PEER_NAME);
+    if (anetFdToString(conn ? conn->fd : -1, ip, ip_len, port, FD_TO_PEER_NAME) == -1) {
+        conn->last_errno = errno;
```

Comment:
Setting `conn->last_errno` would not work in the case where we have no connection and use fd=-1.
`connPeerToString` is used from a couple of functions, so there might be cases where this is possible.

Reply:
You're right, I haven't notice the possibility of no connection in some other callings. I'll add a check before assignment.